### PR TITLE
Add new zeus addObjects setting to settings module

### PR DESCRIPTION
**Pull request by alganthe**
The module part in CfgVehicles.hpp was missing and thus the option to turn it  on / off wasn't here.

### CODE REVIEW

File: addons/zeus/ACE_Settings.hpp [hunk 1]
```diff
@@ -21,9 +21,7 @@ class ACE_Settings {
         values[] = {"$STR_A3_OPTIONS_DISABLED", CSTRING(revealMines_partial), CSTRING(revealMines_full)};
     };
     class GVAR(autoAddObjects) {
-        displayName = CSTRING(AddObjectsToCurator);
```
Comment (Glowbal) [inline addons/zeus/ACE_Settings.hpp]: Why are these removed?

**Comment by alganthe** [comment]
The other settings don't have those lines (reveal mines and ascension) thus i though it wasn't necessary, should i add it back?

**Comment by Glowbal** [comment]
Yes they should be added back. It is for the Config Export menu.

**Comment by alganthe** [comment]
adding all of them now, note that some other modules are also missing them

**Comment by SilentSpike** [comment]
Will have to give this one a think, this setting might not be suitable for the module.

Potentially we could retroactively add all existing objects once settings are initialized. However this wouldn't work out if it was enabled by default and the module tried to disable it.

**Comment by jonpas** [comment]
> Potentially we could retroactively add all existing objects once settings are initialized. However this wouldn't work out if it was enabled by default and the module tried to disable it.

`runAtSettingsInitialized`

**Comment by Glowbal** [comment]
This PR adds unrelated code as well. That should probably be in a separate PR.

**Comment by Glowbal** [comment]
If we change this to use `runAtSettingsInitialized` we can merge this, I think.

**Comment by SilentSpike** [comment]
My memory needs a refresher: `runAtSettingsInitialized` = run immediately if they have initialized, but wait if they haven't? (The naming is a little ambiguous)

**Comment by jonpas** [comment]
@SilentSpike That's up to the function, `runAtSettingsInitialized` simply adds the function to an array for later execution, so you can relay on eg. module settings in the function and based on the setting do or don't do something.

It is not based on specific setting, the functions which are waiting for settings to initialize using this will always run, it's up to the function to exit if it shouldn't do something based on settings.

**Comment by SilentSpike** [comment]
Wait, what? :confused:

I think that's a yes :laughing:

**Comment by jonpas** [comment]
I think I misunderstood you a bit. :grin: 

If I understand what you are asking now, then I don't know really, the correct logic for `runAtSettingsInitialized` is https://github.com/acemod/ACE3/blob/master/addons/repair/functions/fnc_addSpareParts.sqf#L29-L31
Which means if settings are already initialized they shouldn't be added to `runAtSettingsInitialized` at all, I guess.

**Comment by PabstMirror** [comment]
variable isn't read in the module sqf:
https://github.com/alganthe/ACE3ganthe/blob/zeus_fix/addons/zeus/functions/fnc_moduleZeusSettings.sqf

**Comment by Glowbal** [comment]
I will take over and finish this up

**Comment by Glowbal** [comment]
Closing in favour of #3768 that continues the work done in this PR.
